zio-schema/shared/src/main/scala/zio/schema/DynamicValue.scala

```diff
@@ -19,29 +19,40 @@ sealed trait DynamicValue {
     }
 
   def toTypedValue[A](implicit schema: Schema[A]): Either[String, A] =
-    toTypedValueLazyError.left.map(_.apply())
+    toTypedValueLazyError.left.map {
+      case DynamicValue.Err.CastError(value, schema) => s"Failed to cast $value to schema $schema"
```

Comment:
Maybe you could expose a `.message` on `Err` so these human-readable strings can be consolidated and located in one (obvious) place?